components/learn/syllabus-tree/syllabus-tree-container.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import styled from 'react-emotion';
+
+import TreeView from './syllabus-tree-component';
+
+export default class SyllabusTree extends React.Component {
+  state = {
+    nodeStateTracker: this.props.data.map(() => true),
+  };
+
+  handleClick = i => {
+    this.setState({
+      nodeStateTracker: [
+        ...this.state.nodeStateTracker.slice(0, i),
+        !this.state.nodeStateTracker[i],
+        ...this.state.nodeStateTracker.slice(i + 1),
+      ],
```

Comment:
I am curious to know if the implementation I mentioned below is right according to React practices
```
handleClick = i => {
const { nodeStateTracker } = this.state;
nodeStateTracker[i] = false;
this.forceUpdate()
}
```

Reply:
Using forceUpdate is often discouraged which can be done with setState instead.
Also state is suppose to be updated with setState . But i havent tried the above code, let me know if its work

Reply:
Yeah I took the assumption that `nodeStateTracker[i]` is a child of the state array hence it may not re-render. Hence I am `forceUpdate`. I kept in my mind that we cannot mutate the state directly. See if this works and let me know.